src/front/glsl/lex_pp.rs

```diff
@@ -0,0 +1,242 @@
+                Punct::ModAssign => Token::ModAssign(meta),
+                Punct::LeftShiftAssign => Token::LeftAssign(meta),
+                Punct::RightShiftAssign => Token::RightAssign(meta),
+                Punct::AndAssign => Token::AndAssign(meta),
+                Punct::XorAssign => Token::XorAssign(meta),
+                Punct::OrAssign => Token::OrAssign(meta),
+
+                // Two character punctuation
+                Punct::Increment => Token::IncOp(meta),
+                Punct::Decrement => Token::DecOp(meta),
+                Punct::LogicalAnd => Token::AndOp(meta),
+                Punct::LogicalOr => Token::OrOp(meta),
+                Punct::LogicalXor => Token::XorOp(meta),
+                Punct::LessEqual => Token::LeOp(meta),
+                Punct::GreaterEqual => Token::GeOp(meta),
+                Punct::EqualEqual => Token::EqOp(meta),
+                Punct::NotEqual => Token::NeOp(meta),
+                Punct::LeftShift => Token::LeftOp(meta),
+                Punct::RightShift => Token::RightOp(meta),
+
+                // Parenthesis or similar
+                Punct::LeftBrace => Token::LeftBrace(meta),
+                Punct::RightBrace => Token::RightBrace(meta),
+                Punct::LeftParen => Token::LeftParen(meta),
+                Punct::RightParen => Token::RightParen(meta),
+                Punct::LeftBracket => Token::LeftBracket(meta),
+                Punct::RightBracket => Token::RightBracket(meta),
+
+                // Other one character punctuation
+                Punct::LeftAngle => Token::LeftAngle(meta),
+                Punct::RightAngle => Token::RightAngle(meta),
+                Punct::Semicolon => Token::Semicolon(meta),
+                Punct::Comma => Token::Comma(meta),
+                Punct::Colon => Token::Colon(meta),
+                Punct::Dot => Token::Dot(meta),
+                Punct::Equal => Token::Equal(meta),
+                Punct::Bang => Token::Bang(meta),
+                Punct::Minus => Token::Dash(meta),
+                Punct::Tilde => Token::Tilde(meta),
+                Punct::Plus => Token::Plus(meta),
+                Punct::Star => Token::Star(meta),
+                Punct::Slash => Token::Slash(meta),
+                Punct::Percent => Token::Percent(meta),
+                Punct::Pipe => Token::VerticalBar(meta),
+                Punct::Caret => Token::Caret(meta),
+                Punct::Ampersand => Token::Ampersand(meta),
+                Punct::Question => Token::Question(meta),
+            },
+            TokenValue::Version(version) => {
+                for t in version.tokens {
+                    self.pp_buf.push_back(t);
+                }
+                Token::Version(meta)
+            }
+            _ => Token::Unknown((meta, format!("{:?}", pp_token.value))),
+        })
+    }
+}
+
+impl<'a> Iterator for LexerPP<'a> {
+    type Item = Token;
+    fn next(&mut self) -> Option<Self::Item> {
+        self.next()
```

Comment:
why do we need `fn next()` as an internal method if we are implementing the iterator anyway? Can we instead just move the logic in here?